app/models/package_manager/npm.rb

```diff
@@ -99,10 +99,15 @@ def self.versions(raw_project, _name)
         license = v.fetch("license", nil)
         license = licenses(v) unless license.is_a?(String)
         license = "" if license.nil?
+
+        # if deprecated field is present and is not false, it can be an arbitrary string, so convert to boolean
```

Comment:
What kinds of arbitrary strings can be in here, and would it be possible for the arbitrary string to contradict the assumption that its presence == deprecated?

Reply:
That's a good question. From my experience just looking at anecdotal examples I've found, it appears that they are essentially deprecation reasons. But I can try to find documentation in the NPM API to confirm that the assumption is correct.

Reply:
Weirdly, I'm having a bizarrely hard time finding any documentation on this field beyond what I have tangibly seen in the response. Here is the API [documentation](https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md#version) for the `Version` type which is the type of the array elements of the `Package.versions` field. I'm not seeing anything about a `deprecated` field even though I have encountered in on some response objects. 

The [README](https://github.com/npm/registry/tree/master) says this is archived documentation so maybe there's a more up-to-date one, though I haven't found that anywhere yet. Will keep looking though, because now I'm a little concerned if we're depending on a field that isn't even specified by the docs.

Reply:
I have been unable to find documentation indicating the exact behavior of the `deprecated` field here.

But I have found several anecdotal examples confirming the behavior appears to be as expected by this code:

If `deprecated` is `false` or not present, the version is not deprecated
Otherwise, `deprecated` has always been a string with a deprecation reason, indicating that yes it is deprecated

I'm inclined to follow this behavior, as it seems that there's not another effective way to confirm this.